vignettes/rhino-fluent.Rmd

```diff
@@ -0,0 +1,560 @@
+So, in this section, we will introduce a table to our application that displays random imaginary sales deals.
+The dataset comes from the `shiny.fluent` package itself - [fluentSalesDeals](https://appsilon.github.io/shiny.fluent/reference/fluentSalesDeals.html).
+
+### Creating the module
+
+In the `app/view/` folder, let us create a new file and name it `datatable.R`
+
+This `app/view/datatable.R` file will be used to display the data table on the UI and we will call this module in the `app/main.R` file once done. Here is the code
+
+```
+# app/view/datatable.R
+
+box::use(
+  shiny[moduleServer, NS, div, uiOutput, renderUI],
+  shiny.fluent[fluentPage, Text, DetailsList]
+)
+
+#' @export
+ui <- function(id) {
+  ns <- NS(id)
+    div(
+      Text("A randomly generated dataset of imaginary sales deals", variant = "large"),
+      uiOutput(ns("datatable"))
+    )
+}
+
+#' @export
+server <- function(id) {
+  moduleServer(id, function(input, output, session) {
+    output$datatable <- renderUI({
+      # Datasets are the only case when you need to use :: in `box`.
+      # This issue should be solved in the next `box` release.
+      DetailsList(items = shiny.fluent::fluentSalesDeals)
+    })
+  })
+}
+```
+
+`fluentSalesDeals` is a dataset available with the `shiny.fluent` package and to use it, we use `::` for directly fetching the dataset from the library. Datasets are the only case when you need to use `::` in `box`.
+
+So, here we are using [`DetailsList`](https://appsilon.github.io/shiny.fluent/reference/DetailsList.html) function in the server side to render the table on the UI.
+
+### Calling the module
+
+Our shiny module `app/view/datatable.R` is ready, but unless we call it in the `app/main.R` file, we won't see any changes on the app. Let us thus go to `app/main.R` file and edit it with the following code -
+
+```         
+# app/main.R
+
+box::use(
+  shiny[moduleServer, NS, div],
+  shiny.fluent[fluentPage, Text]
+)
+
+box::use(
+  app/view/datatable
+)
+
+#' @export
+ui <- function(id) {
+  ns <- NS(id)
+  fluentPage(      
+    Text("Sales Data Dashboard", variant = "xxLarge"),
+    datatable$ui(ns("datatable"))
+  )
+}
+
+#' @export
+server <- function(id) {
+  moduleServer(id, function(input, output, session) {
+    datatable$server("datatable")
+  })
+}
+```
+
+This code in the `app/main.R` file is based the concept of [Shiny modules](https://shiny.rstudio.com/articles/modules.html) where [box modules in rhino](https://appsilon.github.io/rhino/articles/explanation/box-modules.html) are used.
+
+Notice here that we have called the `datatable` module by importing it to our main application file first.
+It is now, that the `main` module will be able to use exported functions from `app/view/datatable.R` module.
+
+Now run the app.
+You should be seeing something like this -
+
+```{r, echo=FALSE}
+knitr::include_graphics("images/rhino_fluent_2.png")
+```
+
+### Some improvements
+
+#### Show informative columns with meaningful names
```

Comment:
Having a section called in a meaningless way: "Some improvements" makes no sense, especially since it contains only one subsection.